File: 30 days of code JAVA/day10.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class hackbin {
    
    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        int co=n;
        int arr[]= new int[40];
        int i=0;
        
      while(co!=0)
       {  arr[i]=co%2;
          i++; 
          co=co/2;
       }
                    
    // for(i=15;i>-1;i--)
     //    System.out.print(arr[i]);
     int c=0;
     int max=0;
     if(arr[0]==1) {max=1; c=1;}
     
     for(i=1;i<=39;i++)
     {
          if((arr[i]==1 && arr[i+1]==1))
               c++;
        else  if (arr[i-1]==1 && arr[i]==1) c++;
          
          else if(arr[i]==0) c=0;   
               else c=1;  
          if(max<c)
             { max=c;}
     
      }

    //System.out.println();
    System.out.print(max);     

        scanner.close();
    }
}

File: 30daysOFcode_Scala/day 10.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._
import scala.collection.mutable.ListBuffer
import scala.collection.mutable.HashSet

object Solution {
    def main(args: Array[String]) {
    
    var count=1
    var max_seq=0
    val hash_set: HashSet[Int] = HashSet.empty[Int]
    val stdin = scala.io.StdIn
    var n = stdin.readLine.trim.toInt
    var list = new ListBuffer[Int]()

    while(n>=1){
       list+=n%2
       n =n/2
    }
    
    list=list.reverse.zipWithIndex.filter(_._1>0).map(x=>(x._2+1))
    
    for (e <- list) hash_set+=e

    for (e <- list){
         if(hash_set.contains(e-1)){
            count=count+1
         }else{
           count=1
         }
         if(count>max_seq){
        max_seq=count
      }
    }
        println(max_seq)
    }
}
